tools/json2csv.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+from pathlib import Path
+from collections import defaultdict
+import json
+
+import numpy as np
+import pandas as pd
+
+
+def get_parser():
+    """Build parser object"""
+    from argparse import ArgumentParser
+    from argparse import RawTextHelpFormatter
+
+    parser = ArgumentParser(description='MRIQC-WebAPI: massaging bson dumps',
+                            formatter_class=RawTextHelpFormatter)
+    parser.add_argument('input_file', action='store', type=Path, help='input')
+    parser.add_argument('output_file', action='store', type=Path, help='output')
+    parser.add_argument('--no-dedup', action='store_true', default=False,
+                        help='do not deduplicate entries')
+    return parser
+
+
+def main():
+    args = get_parser().parse_args()
+
+    dictlist = []
+    with args.input_file.open() as f:
+        for entry in f.readlines():
+            entry = entry.strip().replace('+Infinity', '"+Infinity"')
+            entry = entry.replace('-Infinity', '"-Infinity"')
+            try:
+                dictlist += [json.loads(entry.strip())]
+            except:
+                print("Error reading: ", entry.strip())
+
+    if args.no_dedup:  # processing ratings
+        data = defaultdict(list, {})
+        for entry in dictlist:
+            data['created'] += [entry['_created']['$date'] or '']
+            data['updated'] += [entry['_updated']['$date'] or '']
+            data['rating'] += [entry['rating']]
+            data['md5sum'] += [entry['md5sum']]
+            data['name'] += [entry['name'] or '']
+            data['comment'] += [entry['comment'] or '']
+
+        pd.DataFrame(data)[['md5sum', 'rating', 'created', 'updated', 'name', 'comment']].to_csv(
+            str(args.output_file), index=None)
+        return 0
+
+
+    def emptylist():
+        """Create an array of NaNs of the same size as currently
+        stored entries"""
+        return [np.nan] * max(0, len(data.get('md5sum', [])) - 1)
+
+    data = defaultdict(emptylist, {})
+    for entry in dictlist:
+        md5sum = entry['provenance']['md5sum']
```

Comment:
dumps from the rating collection don't have a provenance field, md5sum should exist at the top level of the object.

Reply:
Yep, you need to use the flag `--no-dedup` for those. Probably the flag could use a renaming.